### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by FL3SH** [containous/traefik#6761]
### Do you want to request a *feature* or report a *bug*?
Bug
### What did you do?
Configure traefik with kubernetsCRD using IngressRoute, IngressRouteTCP, and IngressRouteUDP.

I think it might be correlated with my IngressRouteTCP.
### What did you expect to see?
Memory usage shouldn't constantly increase until pod is killed by Kubernetes.

### What did you see instead?
Constantly increasing memory usage.

<img width="1602" alt="Screenshot 2020-05-02 at 14 54 42" src="https://user-images.githubusercontent.com/9084725/80869070-84dcb680-8c9e-11ea-8164-1050f04bf6ef.png">

 If necessary I can post more screenshots from Prometheus.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
image: traefik:2.2.1 (same behavior for 2.2.0)
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Kubernetes CRD
```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRouteTCP
metadata:
  name: mqtt
  namespace: default
spec:
  entryPoints:
  - mqtt
  routes:
  - match: HostSNI(`*`)
    services:
    - name: mosquitto
      port: 8883
---
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRouteUDP
metadata:
  name: radius1-udp
  namespace: default
spec:
  entryPoints:
  - radius1
  routes:
  - services:
    - name: myapp-udp
      port: 1812
---
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRouteUDP
metadata:
  name: radius2-udp
  namespace: default
spec:
  entryPoints:
  - radius2
  routes:
  - services:
    - name: myapp-udp
      port: 1813
```

IngressRoutes are quite long, let me know if I need to post them too.

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)
I think only applicable log that I am seeing:
```
msg="Error during connection: readfrom tcp 10.128.2.20:47420->10.128.4.15:8883: read tcp 10.128.2.20:8883->202.130.98.126:29192: read: connection reset by peer"
```

**Comment by FL3SH** [comment]
After some testing, I determined that problem is in handing UDP.
After removing IngressRouteUDP memory leak disappear.
You can easily reproduce this issue by sending UDP packest ex. using bash:
```bash
for i in `seq 1 100000`; do echo "test" > /dev/udp/<treafik_IP>/1812; sleep 1; echo $i; done
```

I thought I found walkaround for this issue (yellow chart) by enabling `tracing.elastic.serverURL`, but after restart problems seems to occur again.

<img width="1613" alt="Screenshot 2020-05-08 at 21 49 10" src="https://user-images.githubusercontent.com/9084725/81443574-f2448780-9175-11ea-8d30-b7048b12f0a0.png">

**Comment by dtomcej** [comment]
Hello @FL3SH,

Could you enable debug mode and the API, and grab the pprof CPU and mem traces and attach them here?

It would assist in debugging what exactly is taking so much memory.

https://docs.traefik.io/operations/api/#debug

Thanks!

**Comment by FL3SH** [comment]
I should definitely start with the docs first, because I build my custom docker from `master (6b1158235e145635d79d251ac1d8a3865258e329)` with `go func() { log.Fatal(http.ListenAndServe("0.0.0.0:3333", nil)) }()` in `cmd/traefik/traefik.go` to capture some heap dumps.

[trace.tar.gz](https://github.com/containous/traefik/files/4603047/trace.tar.gz)
[profile.tar.gz](https://github.com/containous/traefik/files/4603048/profile.tar.gz)
[heap.tar.gz](https://github.com/containous/traefik/files/4603049/heap.tar.gz)

Every archive have to files when ram usage was around 300mb and almost 2gb.

Please let me know if dumps are ok, or do I need to redo it using traefik api debug options.

**Comment by FL3SH** [comment]
Do you need more heap dumps or can I stop dumping it?

**Comment by dtomcej** [comment]
Hi @FL3SH I think one should be sufficient, thanks!

**Comment by FL3SH** [comment]
Perfect, please keep me updated.

**Comment by ddtmachado** [comment]
@FL3SH Looked at your dump / profile and it seemed that Traefik was leaking memory and go routines while allocating new UDP connection handlers.

Also I was able to reproduce it, in a similar fashion as your example, by starting a lot of new connections (from different source ip/ports) in a short span of time (that's because Traefik will automatically close the connection after 3 seconds of inactivity).

A PR to fix it is opened and waiting for the review of a senior maintainer.

**Comment by FL3SH** [comment]
Thank you ;)

**Comment by mpl** [comment]
@FL3SH have you tried @ddtmachado 's PR, and do you confirm that it seems to fix the problem for you?

**Comment by FL3SH** [comment]
Yes, looks like @ddtmachado 's traefik works perfectly, even ram usage is lowest than in 2.2.1, but as I mentioned in issue this problem didn't occur always, so I will update the issue if it occurs, right now for me, it is solved.

**Comment by FL3SH** [comment]
It's been running more than two days without any issues.

**Comment by jbdoumenjou** [comment]
closed by #6815
